src/Npgsql/Internal/PgTypeInfo.cs

```diff
@@ -114,24 +114,27 @@ internal PgConverterResolution GetResolution()
         return new(Converter, PgTypeId.GetValueOrDefault());
     }
 
-    bool CachedCanConvert(DataFormat format, out BufferRequirements bufferRequirements)
+    bool CanConvert(PgConverter converter, DataFormat format, out BufferRequirements bufferRequirements)
     {
-        if (format is DataFormat.Binary)
+        if (HasCachedInfo(converter))
         {
-            bufferRequirements = _binaryBufferRequirements;
-            return _canBinaryConvert;
+            switch (format)
+            {
+            case DataFormat.Binary:
+                bufferRequirements = _binaryBufferRequirements;
+                return _canBinaryConvert;
+            case DataFormat.Text:
+                bufferRequirements = _textBufferRequirements;
+                return _canTextConvert;
+            }
```

Comment:
nit: add default and throw

Reply:
We fall through instead and let the converter decide.